entanglement_forging/utils/forging_subroutines.py

```diff
@@ -142,24 +174,41 @@ def eval_forged_op_with_result(
     pauli_vals_tensor_states_extrap = richardson_extrapolate(
         pauli_vals_tensor_states_raw, richardson_stretch_factors, axis=2
     )
-    superpos_state_prefixes = []
+
+    superpos_state_prefixes_u = []
+    superpos_state_prefixes_v = []
     superpos_state_indices = []
     lin_combos = ["xplus", "xmin"]  # ,'yplus','ymin']
-    num_bitstrings = len(bitstrings_s)
+    # num_bitstrings is the number of bitstring combos we have
+    num_bitstrings = len(bitstrings_s_u)
     for x in range(num_bitstrings):
         for y in range(num_bitstrings):
             if x == y:
                 continue
-            bs_string = f"bs{min(x,y)}bs{max(x,y)}"
-            superpos_state_prefixes += [
-                bs_string + lin_combo for lin_combo in lin_combos
+            bsu_string = f"bsu{min(x,y)}bsu{max(x,y)}"
+            superpos_state_prefixes_u += [
+                bsu_string + lin_combo for lin_combo in lin_combos
             ]
+
+            # Determine whether we are handling the two subsystems separately
+            asymmetric_bitstrings = False
```

Comment:
I guess we can keep using bitstrings_v to determine this case. No need for an extra variable